src/linear_samplers/block_row_rand_cyclic.jl

```diff
@@ -0,0 +1,89 @@
+# This code was Written by Nathaniel Pritchard
+"""
+    LinSysBlkRowRandCyclic <: LinSysBlkRowSampler
+
+A mutable structure with fields to handle randomly permuted block sampling. Can allow for fixed
+blocks or blocks whose entries are randomly permuted. After each cycle, a new random ordering is
+created. If the last block would be smaller than the others, rows from a previous block are 
+added to keep all blocks the same size.
+
+# Fields
+- `blockSize::Int64` - Specifies the size of each block.
+- `constantBlock::Bool` - A variable specifying if the user would like for the rows to be randomly
+permuted. 
+- `nBlocks::Int64` - Variable that contains the number of blocks overall.
+- `order::Vector{Int64}` - The order that the blocks will be used to generate updates.
+- `blocks::Vector{Int64}` - The list of all the row in each block.
+
+Calling `LinSysVecColBlockRandCyclic()` defaults to setting `blockSize` to 2 and `constantBlock` to true. The `sample`
+function will handle the re-initialization of the fields once the system is provided.
+"""
+mutable struct LinSysBlkRowRandCyclic <: LinSysBlkRowSampler
+    blockSize::Int64
+    constantBlock::Bool
+    nBlocks::Int64
+    order::Vector{Int64}
+    blocks::Vector{Int64}
+end
+
+LinSysBlkRowRandCyclic(blockSize, constantBlock) = LinSysBlkRowRandCyclic(blockSize, constantBlock, 1, Int64[], Int64[])
+LinSysBlkRowRandCyclic(blockSize) = LinSysBlkRowRandCyclic(blockSize, true, 1, Int64[], Int64[])
+LinSysBlkRowRandCyclic() = LinSysBlkRowRandCyclic(2, true, 1,  Int64[], Int64[])
+
+# Common sample interface for linear systems
+function sample(
+    type::LinSysBlkRowRandCyclic,
+    A::AbstractArray,
+    b::AbstractVector,
+    x::AbstractVector,
+    iter::Int64
+)
+    if iter == 1
+        m, n = size(A)
+        # Determine number of blocks
+        type.nBlocks = div(m, type.blockSize) + (rem(m, type.blockSize) == 0 ? 0 : 1)
+        blockIdxs = type.blockSize * type.nBlocks
+        lastBlockStart = blockIdxs - type.blockSize + 1 
+        type.blocks = Vector{Int64}(undef, blockIdxs)
+        # Block definitions
+        if type.constantBlock
+            type.blocks[1:lastBlockStart - 1] .= collect(1:lastBlockStart - 1)
+            if rem(n, type.blockSize) == 0
+                type.blocks[lastBlockStart:blockIdxs] .= collect(lastBlockStart:m)
+            else
+                # maintain size of last block using last blockSize rows 
+                type.blocks[lastBlockStart:blockIdxs] .= collect(vcat(m - type.blockSize + 1:lastBlockStart - 1, lastBlockStart:m))
+            end
+        
```

Comment:
@npritch928 Is this a repetition from the previous block size code? If so, can we have a function to implement this.

Reply:
@dmaldona I think this is a good point. But it may be worth letting this go for the current version. I think v0.2 might consider how we may want to reuse code a bit better. Or do you think we should do it now?